hadoop-hdds/server-scm/src/test/java/org/apache/hadoop/hdds/scm/node/TestNodeDecommissionManager.java

```diff
@@ -369,10 +410,208 @@ public void testNodeDecommissionManagerOnBecomeLeader() throws Exception {
     assertEquals(decom.getMonitor().getTrackedNodes().size(), 3);
   }
 
-  private SCMNodeManager createNodeManager(OzoneConfiguration config)
-      throws IOException, AuthenticationException {
-    scm = HddsTestUtils.getScm(config);
-    return (SCMNodeManager) scm.getScmNodeManager();
+  @Test
+  public void testInsufficientNodeDecommissionThrowsExceptionForRatis() throws
```

Comment:
The newly added test cases look good. Can we also assert that the DNs have transitioned to the decommissioning state wherever relevant? Existing tests are doing that.

Reply:
Thank you for the review! I have added these checks in the tests. I have addressed the other comments as well. Could you please take a look at this PR again and approve the workflows if everything seems okay?